types/wlr_keyboard_shortcuts_inhibit_v1.c

```diff
@@ -0,0 +1,212 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <util/signal.h>
+#include <wlr/types/wlr_keyboard_shortcuts_inhibit_v1.h>
+#include "keyboard-shortcuts-inhibit-unstable-v1-protocol.h"
+
+static const struct zwp_keyboard_shortcuts_inhibit_manager_v1_interface
+	keyboard_shortcuts_inhibit_impl;
+
+static const struct zwp_keyboard_shortcuts_inhibitor_v1_interface
+	keyboard_shortcuts_inhibitor_impl;
+
+static struct wlr_keyboard_shortcuts_inhibit_manager_v1 *
+wlr_keyboard_shortcuts_inhibit_manager_v1_from_resource(
+		struct wl_resource *resource) {
+	assert(wl_resource_instance_of(resource,
+		&zwp_keyboard_shortcuts_inhibit_manager_v1_interface,
+		&keyboard_shortcuts_inhibit_impl));
+	return wl_resource_get_user_data(resource);
+}
+
+static struct wlr_keyboard_shortcuts_inhibitor_v1 *
+wlr_keyboard_shortcuts_inhibitor_v1_from_resource(
+		struct wl_resource *resource) {
+	assert(wl_resource_instance_of(resource,
+		&zwp_keyboard_shortcuts_inhibitor_v1_interface,
+		&keyboard_shortcuts_inhibitor_impl));
+	return wl_resource_get_user_data(resource);
+}
+
+static void keyboard_shortcuts_inhibitor_v1_destroy(
+		struct wlr_keyboard_shortcuts_inhibitor_v1 *inhibitor) {
+	if (!inhibitor) {
+		return;
+	}
+
+	wlr_signal_emit_safe(&inhibitor->events.destroy, inhibitor);
+
+	wl_resource_set_user_data(inhibitor->resource, NULL);
+	wl_list_remove(&inhibitor->manager_link);
+	wl_list_remove(&inhibitor->surface_destroy.link);
+	wl_list_remove(&inhibitor->seat_client_destroy.link);
+	free(inhibitor);
+}
+
+static void keyboard_shortcuts_inhibitor_v1_handle_resource_destroy(
+		struct wl_resource *resource) {
+	struct wlr_keyboard_shortcuts_inhibitor_v1 *inhibitor =
+		wlr_keyboard_shortcuts_inhibitor_v1_from_resource(resource);
+	keyboard_shortcuts_inhibitor_v1_destroy(inhibitor);
+}
+
+static void keyboard_shortcuts_inhibitor_handle_surface_destroy(
+		struct wl_listener *listener, void *data) {
+	struct wlr_keyboard_shortcuts_inhibitor_v1 *inhibitor =
+		wl_container_of(listener, inhibitor, surface_destroy);
+	keyboard_shortcuts_inhibitor_v1_destroy(inhibitor);
+}
+
+static void keyboard_shortcuts_inhibitor_handle_seat_client_destroy(
+		struct wl_listener *listener, void *data) {
+	struct wlr_keyboard_shortcuts_inhibitor_v1 *inhibitor =
+		wl_container_of(listener, inhibitor, seat_client_destroy);
+	keyboard_shortcuts_inhibitor_v1_destroy(inhibitor);
+}
+
+static void keyboard_shortcuts_inhibitor_v1_handle_destroy(
+		struct wl_client *client,
+		struct wl_resource *manager_resource) {
+	wl_resource_destroy(manager_resource);
```

Comment:
This name is misleading, since it's not a manager resource, it's an inhibitor resource. We should rename this variable to `resource` or `inhibitor_resource`.

Reply:
Done.